Upgrade to Node 20 and add outputs

jgillick opened the pull request:
* Github Actions is deprecating Node 12.
* Added the following outputs:
   * `found-artifact`: `boolean`
   * `path `: `string`
 * Used `ncc` to bundle the script and dependencies into a single JS file (`dist/index/js`), so that `node_modules` doesn't need to be committed to github.

Now, every time `main.js` is updated, you'll need to run `npm script build` to generate the compiled script. If you prefer bundling `node_modules`, we can revert this change. However, committing `node_modules` makes PRs pretty messy because they'll include every file that changed in all of the modules.

jgillick committed:
Output artifact download path

jgillick committed:
Upgrade node. Bump version number

jgillick committed:
Upgrade actions/setup-node to v3

jgillick committed:
Upgrade node

jgillick committed:
Update node libraries

jgillick committed:
Compile action into single file. (no longer commit node_modules)

jgillick committed:
Add node_modules to gitignore

jgillick committed:
Fix filesize import

aochmann commented:
@jgillick thanks for submit PR - I will review and come to you with update. BTW. sorry for so late response 🙏🏻

ardittristan commented:
Github is now deprecating node16 and using node20

jgillick committed:
Cleanup IDE formatting and package installs

jgillick committed:
Set node version to 20

jgillick committed:
Update built file

jgillick commented:
I've updated the PR to node20 and removed the automatic formatting my IDE applied to `main.js`. I also updated the PR description explaining why `node_modules` has been removed.

jgillick commented:
If anybody else want to use this before the PR is merged, you can update your YML files to use `jgillick/actions-download-artifact@master`

jgillick committed:
Fix filesize import
